### DISCUSSION TITLE
Accessibility - Batch-Adding Translucent Black Background to Subtitles

### DISCUSSION THREAD

**Discussion by veryprogram** [ImageMagick/ImageMagick#6441] [Help]
### ImageMagick version

20.1.0

### Operating system, version and so on

macOS Ventura 13.4

### The Situation
I have a bunch of .MKV files — Blu-Ray rips — that contain PGS subtitles. (Subtitles that are pictures of text, rather than _text_ text.) These subtitles are white text only, with no background, and as such they are difficult for people with poor vision to see; adding a translucent black background would fix the legibility problem. I have figured out how to export PGS subtitles as a PNG transparent image sequence (using BDSup2Sup). These resulting images are white text with no background, and they are of varying height and width; they are sequentially named 0001.png, 0002.png, 0003.png, and so on.

(Yes, I know I could just use OCR to convert these subtitles to true text and then configure subtitle styling the usual way, but OCR introduces so many errors that require manual correction so as to be unfeasible for so many files. Modifying the pictures of text is the much more feasible, scaleable path.)

### The Solution
I need a way to loop through all these four digit sequentially-numbered transparent PNGs and add a translucent black background to each of them. I think the best way to do this would be to make a separate, singular transparent PNG file that is just a translucent black rectangle, then use ImageMagick's Composite action to add the subtitle PNG on top of the rectangle — but I don't know how to make ImageMagick automatically stretch that transparent PNG file of a translucent black rectangle in order to match each subtitle PNG's dimensions, and for that matter I still haven't figured out how to composite a subtitle PNG with a translucent black rectangle PNG that already matches its dimensions. I've been reading the docs, but I'm still at a loss.

I'm definitely not the only person who's encountered the problem of PGS subtitles being insufficiently legible while not having the time to manually error-correct OCR, so if we can figure this out, I hope those other people end up being able to find this thread in the future. :)

**Comment by fmw42** [comment]
Please post and example PNG that you have and need to add a translucent background.  Are all your PNG files the same dimensions?

**Comment by veryprogram** [comment]
Example subtitle 1:
![0001](https://github.com/ImageMagick/ImageMagick/assets/93247202/6113753b-c61b-47ec-9400-4d3df57acd0b)

Example subtitle 2:
![0024](https://github.com/ImageMagick/ImageMagick/assets/93247202/10299679-a4bc-420c-a4d4-43a4e33cee57)

This is what they look like, and unfortunately they're all different dimensions, hence wondering whether it's possible for ImageMagick to stretch the background to match each individual PNG.

**Reply by fmw42** [reply]
Does this do what you want?

`magick text.png -background "rgba(0,0,0,0.125)" -flatten x.png`

![text](https://github.com/ImageMagick/ImageMagick/assets/13458218/4b6ce714-a026-494b-8e75-fe97cfb54aea)


or if you want color

`magick text.png -background "rgba(0,255,255,0.125)" -flatten y.png`

![y](https://github.com/ImageMagick/ImageMagick/assets/13458218/2df56ab3-3537-445c-96a5-585fd2c8d3d3)

Adjust the alpha value in the rgba setting for the amount of opacity (larger values -- closer to 1 -- will be more opaque and values closer to 0 will be more transparent)

**Reply by veryprogram** [reply]
Ooh yes, that's great! Do you know whether there's a way to add padding on the outside? Say, like 10 pixels on each side?

**Reply by fmw42** [reply]
Here is 20 px padding

`magick text.png -background "rgba(0,0,0,0.125)" -flatten -compose copy -bordercolor "rgba(0,0,0,0.125)" -border 20 x.png`

![x](https://github.com/ImageMagick/ImageMagick/assets/13458218/2a505da1-0f9c-416c-b231-717cfec6dcbe)

**Reply by veryprogram** [reply]
You're amazing. Thank you so much!
